## 필수과제들
- **필수과제1**
-  q1 - pclass 3 이고, age 20세 미만인 남성과, 여성의 카운트 수와 생존수를 비교해 주세요.
    - 코드와 데이터 값

- **필수과제2**
- 요금에 따라 3가지 등급으로 나눠주시고 (요금의 등급의 기준은 기초통계치 분위수 값으로 기준)
    - 해당 분위수에 따라 남성과 여성을 추출하고 카운팅 수, 생존수 비교해 주세요.
    
- **필수과제3**  
- 혼자온 사람과 혼자 오지 않은사람들 중에서 둘을 비교하는데
    - 공통 조건은 혼자 온 사람의 성별 여부에 따른 전체 카운팅과 생존수 비교해 주세요!
    
    
- 필수과제 요구조건
    - 쥬피터노트북, 코드와 주석, 데이터 비교값을 공유

In [3]:
import seaborn as sns

In [5]:
tt=sns.load_dataset('titanic')

In [7]:
tt

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
886,0,2,male,27.0,0,0,13.0000,S,Second,man,True,NaN,Southampton,no,True
887,1,1,female,19.0,0,0,30.0000,S,First,woman,False,B,Southampton,yes,True
888,0,3,female,NaN,1,2,23.4500,S,Third,woman,False,NaN,Southampton,no,False
889,1,1,male,26.0,0,0,30.0000,C,First,man,True,C,Cherbourg,yes,True


In [12]:
tt_alone=tt[(tt['sibsp']==0)]

In [14]:
tt_alone

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True
5,0,3,male,NaN,0,0,8.4583,Q,Third,man,True,NaN,Queenstown,no,True
6,0,1,male,54.0,0,0,51.8625,S,First,man,True,E,Southampton,no,True
8,1,3,female,27.0,0,2,11.1333,S,Third,woman,False,NaN,Southampton,yes,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
885,0,3,female,39.0,0,5,29.1250,Q,Third,woman,False,NaN,Queenstown,no,False
886,0,2,male,27.0,0,0,13.0000,S,Second,man,True,NaN,Southampton,no,True
887,1,1,female,19.0,0,0,30.0000,S,First,woman,False,B,Southampton,yes,True
889,1,1,male,26.0,0,0,30.0000,C,First,man,True,C,Cherbourg,yes,True


In [30]:
tt_alone_male=tt_alone.query("sex == 'male'")

In [32]:
tt_alone_male

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True
5,0,3,male,NaN,0,0,8.4583,Q,Third,man,True,NaN,Queenstown,no,True
6,0,1,male,54.0,0,0,51.8625,S,First,man,True,E,Southampton,no,True
12,0,3,male,20.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True
17,1,2,male,NaN,0,0,13.0000,S,Second,man,True,NaN,Southampton,yes,True
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
883,0,2,male,28.0,0,0,10.5000,S,Second,man,True,NaN,Southampton,no,True
884,0,3,male,25.0,0,0,7.0500,S,Third,man,True,NaN,Southampton,no,True
886,0,2,male,27.0,0,0,13.0000,S,Second,man,True,NaN,Southampton,no,True
889,1,1,male,26.0,0,0,30.0000,C,First,man,True,C,Cherbourg,yes,True


In [44]:
tt_alone_male_survived=tt_alone_male[tt_alone_male['survived']==1]

In [46]:
tt_alone_male_survived

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
17,1,2,male,NaN,0,0,13.0000,S,Second,man,True,NaN,Southampton,yes,True
21,1,2,male,34.0,0,0,13.0000,S,Second,man,True,D,Southampton,yes,True
23,1,1,male,28.0,0,0,35.5000,S,First,man,True,A,Southampton,yes,True
36,1,3,male,NaN,0,0,7.2292,C,Third,man,True,NaN,Cherbourg,yes,True
55,1,1,male,NaN,0,0,35.5000,S,First,man,True,C,Southampton,yes,True
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
828,1,3,male,NaN,0,0,7.7500,Q,Third,man,True,NaN,Queenstown,yes,True
838,1,3,male,32.0,0,0,56.4958,S,Third,man,True,NaN,Southampton,yes,True
839,1,1,male,NaN,0,0,29.7000,C,First,man,True,C,Cherbourg,yes,True
857,1,1,male,51.0,0,0,26.5500,S,First,man,True,E,Southampton,yes,True


In [48]:
tt_alone_male_notsurvived=tt_alone_male[tt_alone_male['survived']==0]

In [50]:
tt_alone_male_notsurvived

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True
5,0,3,male,NaN,0,0,8.4583,Q,Third,man,True,NaN,Queenstown,no,True
6,0,1,male,54.0,0,0,51.8625,S,First,man,True,E,Southampton,no,True
12,0,3,male,20.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True
20,0,2,male,35.0,0,0,26.0000,S,Second,man,True,NaN,Southampton,no,True
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
881,0,3,male,33.0,0,0,7.8958,S,Third,man,True,NaN,Southampton,no,True
883,0,2,male,28.0,0,0,10.5000,S,Second,man,True,NaN,Southampton,no,True
884,0,3,male,25.0,0,0,7.0500,S,Third,man,True,NaN,Southampton,no,True
886,0,2,male,27.0,0,0,13.0000,S,Second,man,True,NaN,Southampton,no,True


In [34]:
tt_alone_female=tt_alone.query("sex == 'female'")

In [36]:
tt_alone_female

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
8,1,3,female,27.0,0,2,11.1333,S,Third,woman,False,NaN,Southampton,yes,False
11,1,1,female,58.0,0,0,26.5500,S,First,woman,False,C,Southampton,yes,True
14,0,3,female,14.0,0,0,7.8542,S,Third,child,False,NaN,Southampton,no,True
15,1,2,female,55.0,0,0,16.0000,S,Second,woman,False,NaN,Southampton,yes,True
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
879,1,1,female,56.0,0,1,83.1583,C,First,woman,False,C,Cherbourg,yes,False
880,1,2,female,25.0,0,1,26.0000,S,Second,woman,False,NaN,Southampton,yes,False
882,0,3,female,22.0,0,0,10.5167,S,Third,woman,False,NaN,Southampton,no,True
885,0,3,female,39.0,0,5,29.1250,Q,Third,woman,False,NaN,Queenstown,no,False


In [52]:
tt_alone_female_survived=tt_alone_female[tt_alone_female['survived']==1]

In [54]:
tt_alone_female_survived

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
8,1,3,female,27.0,0,2,11.1333,S,Third,woman,False,NaN,Southampton,yes,False
11,1,1,female,58.0,0,0,26.5500,S,First,woman,False,C,Southampton,yes,True
15,1,2,female,55.0,0,0,16.0000,S,Second,woman,False,NaN,Southampton,yes,True
19,1,3,female,NaN,0,0,7.2250,C,Third,woman,False,NaN,Cherbourg,yes,True
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
865,1,2,female,42.0,0,0,13.0000,S,Second,woman,False,NaN,Southampton,yes,True
875,1,3,female,15.0,0,0,7.2250,C,Third,child,False,NaN,Cherbourg,yes,True
879,1,1,female,56.0,0,1,83.1583,C,First,woman,False,C,Cherbourg,yes,False
880,1,2,female,25.0,0,1,26.0000,S,Second,woman,False,NaN,Southampton,yes,False


In [56]:
tt_alone_female_notsurvived=tt_alone_female[tt_alone_female['survived']==0]

In [58]:
tt_alone_female_notsurvived

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
14,0,3,female,14.0,0,0,7.8542,S,Third,child,False,NaN,Southampton,no,True
100,0,3,female,28.0,0,0,7.8958,S,Third,woman,False,NaN,Southampton,no,True
114,0,3,female,17.0,0,0,14.4583,C,Third,woman,False,NaN,Cherbourg,no,True
140,0,3,female,NaN,0,2,15.2458,C,Third,woman,False,NaN,Cherbourg,no,False
177,0,1,female,50.0,0,0,28.7125,C,First,woman,False,C,Cherbourg,no,True
199,0,2,female,24.0,0,0,13.0000,S,Second,woman,False,NaN,Southampton,no,True
205,0,3,female,2.0,0,1,10.4625,S,Third,child,False,G,Southampton,no,False
235,0,3,female,NaN,0,0,7.5500,S,Third,woman,False,NaN,Southampton,no,True
246,0,3,female,25.0,0,0,7.7750,S,Third,woman,False,NaN,Southampton,no,True
254,0,3,female,41.0,0,2,20.2125,S,Third,woman,False,NaN,Southampton,no,False


In [60]:
tt_alone_female_notsurvived.count()

survived       37
pclass         37
sex            37
age            29
sibsp          37
parch          37
fare           37
embarked       37
class          37
who            37
adult_male     37
deck            3
embark_town    37
alive          37
alone          37
dtype: int64

In [66]:
tt_accompanied=tt[(tt['sibsp']>=1)]

In [68]:
tt_accompanied

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
7,0,3,male,2.0,3,1,21.0750,S,Third,child,False,NaN,Southampton,no,False
9,1,2,female,14.0,1,0,30.0708,C,Second,child,False,NaN,Cherbourg,yes,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
866,1,2,female,27.0,1,0,13.8583,C,Second,woman,False,NaN,Cherbourg,yes,False
869,1,3,male,4.0,1,1,11.1333,S,Third,child,False,NaN,Southampton,yes,False
871,1,1,female,47.0,1,1,52.5542,S,First,woman,False,D,Southampton,yes,False
874,1,2,female,28.0,1,0,24.0000,C,Second,woman,False,NaN,Cherbourg,yes,False


In [70]:
tt_accompanied_male=tt_accompanied[tt_accompanied['sex']=='male']

In [72]:
tt_accompanied_male

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
7,0,3,male,2.0,3,1,21.0750,S,Third,child,False,NaN,Southampton,no,False
13,0,3,male,39.0,1,5,31.2750,S,Third,man,True,NaN,Southampton,no,False
16,0,3,male,2.0,4,1,29.1250,Q,Third,child,False,NaN,Queenstown,no,False
27,0,1,male,19.0,3,2,263.0000,S,First,man,True,C,Southampton,no,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
846,0,3,male,NaN,8,2,69.5500,S,Third,man,True,NaN,Southampton,no,False
850,0,3,male,4.0,4,2,31.2750,S,Third,child,False,NaN,Southampton,no,False
860,0,3,male,41.0,2,0,14.1083,S,Third,man,True,NaN,Southampton,no,False
861,0,2,male,21.0,1,0,11.5000,S,Second,man,True,NaN,Southampton,no,False


In [74]:
tt_accompanied_male_survived=tt_accompanied_male[tt_accompanied_male['survived']==1]

In [76]:
tt_accompanied_male_survived

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
65,1,3,male,NaN,1,1,15.2458,C,Third,man,True,NaN,Cherbourg,yes,False
125,1,3,male,12.00,1,0,11.2417,C,Third,child,False,NaN,Cherbourg,yes,False
183,1,2,male,1.00,2,1,39.0000,S,Second,child,False,F,Southampton,yes,False
193,1,2,male,3.00,1,1,26.0000,S,Second,child,False,F,Southampton,yes,False
224,1,1,male,38.00,1,0,90.0000,S,First,man,True,C,Southampton,yes,False
248,1,1,male,37.00,1,1,52.5542,S,First,man,True,D,Southampton,yes,False
261,1,3,male,3.00,4,2,31.3875,S,Third,child,False,NaN,Southampton,yes,False
267,1,3,male,25.00,1,0,7.7750,S,Third,man,True,NaN,Southampton,yes,False
301,1,3,male,NaN,2,0,23.2500,Q,Third,man,True,NaN,Queenstown,yes,False
305,1,1,male,0.92,1,2,151.5500,S,First,child,False,C,Southampton,yes,False


In [78]:
tt_accompanied_male_survived.count()

survived       36
pclass         36
sex            36
age            33
sibsp          36
parch          36
fare           36
embarked       36
class          36
who            36
adult_male     36
deck           18
embark_town    36
alive          36
alone          36
dtype: int64

In [80]:
tt_accompanied_male_notsurvived=tt_accompanied_male[tt_accompanied_male['survived']==0]

In [82]:
tt_accompanied_male_notsurvived

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
7,0,3,male,2.0,3,1,21.0750,S,Third,child,False,NaN,Southampton,no,False
13,0,3,male,39.0,1,5,31.2750,S,Third,man,True,NaN,Southampton,no,False
16,0,3,male,2.0,4,1,29.1250,Q,Third,child,False,NaN,Queenstown,no,False
27,0,1,male,19.0,3,2,263.0000,S,First,man,True,C,Southampton,no,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
824,0,3,male,2.0,4,1,39.6875,S,Third,child,False,NaN,Southampton,no,False
846,0,3,male,NaN,8,2,69.5500,S,Third,man,True,NaN,Southampton,no,False
850,0,3,male,4.0,4,2,31.2750,S,Third,child,False,NaN,Southampton,no,False
860,0,3,male,41.0,2,0,14.1083,S,Third,man,True,NaN,Southampton,no,False


In [84]:
tt_accompanied_female=tt_accompanied[tt_accompanied['sex']=='female']

In [86]:
tt_accompanied_female

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
9,1,2,female,14.0,1,0,30.0708,C,Second,child,False,NaN,Cherbourg,yes,False
10,1,3,female,4.0,1,1,16.7000,S,Third,child,False,G,Southampton,yes,False
18,0,3,female,31.0,1,0,18.0000,S,Third,woman,False,NaN,Southampton,no,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
863,0,3,female,NaN,8,2,69.5500,S,Third,woman,False,NaN,Southampton,no,False
866,1,2,female,27.0,1,0,13.8583,C,Second,woman,False,NaN,Cherbourg,yes,False
871,1,1,female,47.0,1,1,52.5542,S,First,woman,False,D,Southampton,yes,False
874,1,2,female,28.0,1,0,24.0000,C,Second,woman,False,NaN,Cherbourg,yes,False


In [88]:
tt_accompanied_female_survived=tt_accompanied_female[tt_accompanied_female['survived']==1]

In [90]:
tt_accompanied_female_survived

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
9,1,2,female,14.0,1,0,30.0708,C,Second,child,False,NaN,Cherbourg,yes,False
10,1,3,female,4.0,1,1,16.7000,S,Third,child,False,G,Southampton,yes,False
25,1,3,female,38.0,1,5,31.3875,S,Third,woman,False,NaN,Southampton,yes,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
849,1,1,female,NaN,1,0,89.1042,C,First,woman,False,C,Cherbourg,yes,False
856,1,1,female,45.0,1,1,164.8667,S,First,woman,False,NaN,Southampton,yes,False
866,1,2,female,27.0,1,0,13.8583,C,Second,woman,False,NaN,Cherbourg,yes,False
871,1,1,female,47.0,1,1,52.5542,S,First,woman,False,D,Southampton,yes,False


In [92]:
tt_accompanied_female_notsurvived=tt_accompanied_female[tt_accompanied_female['survived']==0]

In [94]:
tt_accompanied_female_notsurvived

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
18,0,3,female,31.0,1,0,18.0000,S,Third,woman,False,NaN,Southampton,no,False
24,0,3,female,8.0,3,1,21.0750,S,Third,child,False,NaN,Southampton,no,False
38,0,3,female,18.0,2,0,18.0000,S,Third,woman,False,NaN,Southampton,no,False
40,0,3,female,40.0,1,0,9.4750,S,Third,woman,False,NaN,Southampton,no,False
41,0,2,female,27.0,1,0,21.0000,S,Second,woman,False,NaN,Southampton,no,False
49,0,3,female,18.0,1,0,17.8000,S,Third,woman,False,NaN,Southampton,no,False
71,0,3,female,16.0,5,2,46.9000,S,Third,woman,False,NaN,Southampton,no,False
111,0,3,female,14.5,1,0,14.4542,C,Third,child,False,NaN,Cherbourg,no,False
113,0,3,female,20.0,1,0,9.8250,S,Third,woman,False,NaN,Southampton,no,False
119,0,3,female,2.0,4,2,31.2750,S,Third,child,False,NaN,Southampton,no,False


In [96]:
tt_accompanied_female_notsurvived.count()

survived       44
pclass         44
sex            44
age            35
sibsp          44
parch          44
fare           44
embarked       44
class          44
who            44
adult_male     44
deck            3
embark_town    44
alive          44
alone          44
dtype: int64

##결괏값
-혼자 온 사람

    남성-434명
        /생존한 남성 73명
        /생존하지 못한 남성 361명
        
    여성-174명
        /생존한 여성 137명
        /생존하지 못한 여성 37명

-같이 온 사람

    남성-143명
        /생존한 남성 36명
        /생존하지 못한 남성 107명

    여성-140명
        /생존한 여성 96명
        /생존하지 못한 여성 44명
    